tests/network/test_continue_response.py

```diff
@@ -0,0 +1,599 @@
+                "params": {
+                    "request": network_id,
+                    "statusCode": -1,
+                },
+            })
+
+
+@pytest.mark.asyncio
+async def test_continue_response_invalid_reason_phrase(websocket, context_id,
+                                                       example_url):
+    network_id = await create_dummy_blocked_request(websocket,
+                                                    context_id,
+                                                    url=example_url,
+                                                    phases=["responseStarted"])
+
+    with pytest.raises(
+            Exception,
+            match=str({
+                "error": "invalid argument",
+                "message": 'Expected string, received array in "reasonPhrase".'
+            })):
+        await execute_command(
+            websocket, {
+                "method": "network.continueResponse",
+                "params": {
+                    "request": network_id,
+                    "reasonPhrase": [],
+                },
+            })
+
+
+@pytest.mark.asyncio
+async def test_continue_response_invalid_headers(websocket, context_id,
+                                                 example_url):
+    network_id = await create_dummy_blocked_request(websocket,
+                                                    context_id,
+                                                    url=example_url,
+                                                    phases=["responseStarted"])
+
+    with pytest.raises(
+            Exception,
+            match=str({
+                "error": "invalid argument",
+                "message": 'Expected array, received string in "headers".'
+            })):
+        await execute_command(
+            websocket, {
+                "method": "network.continueResponse",
+                "params": {
+                    "request": network_id,
+                    "headers": "",
+                },
+            })
+
+
+@pytest.mark.asyncio
+async def test_continue_response_non_blocked_request(websocket, context_id,
+                                                     assert_no_events_in_queue,
+                                                     hang_url):
+    await subscribe(websocket, [
+        "network.beforeRequestSent", "cdp.Network.responseReceived",
+        "cdp.Network.loadingFailed"
+    ])
+
+    await send_JSON_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "context": context_id,
+                "url": hang_url,
+                "wait": "complete",
+            }
+        })
+
+    before_request_sent_event = await wait_for_event(
+        websocket, "network.beforeRequestSent")
+
+    # Assert these events never happen, otherwise the test is ineffective.
+    await assert_no_events_in_queue(
+        ["cdp.Network.responseReceived", "cdp.Network.loadingFailed"],
+        timeout=1.0)
```

Comment:
If the events happen, they do BEFORE the navigation completed. So it would be enough to check that first the `network.beforeRequestSent` event is received, and the navigation command completed right after it. This would allow avoiding adding timeout of 1 sec in the tests.

Reply:
This is important because it covers the case of an invalid server.

For example: test it with localhost:4433 but with no server listening in that port.

This assertion adds a bit of overhead but it is meant to eliminate flakiness.